src/backend/instagram/InstagramConnection.ts

```diff
@@ -0,0 +1,39 @@
+import got, { HTTPError } from 'got';
+
+import { trackConnectionState } from '../utilities/trackConnectionState';
+import { logger } from '../utilities/logger';
+import { configuration } from '../utilities/configuration';
+
+import { instagramEndpoints } from './instagramEndpoints';
+
+export const instagramConnectionState = trackConnectionState(10 * 60 * 1000);
+
+export async function canAccessInstagram() {
+  try {
+    await got
+      .post(instagramEndpoints.token, {
+        form: {
+          client_id: configuration.instagram.clientId,
+          client_secret: configuration.instagram.secret,
+        },
+      })
+      .json();
+    instagramConnectionState.on();
+  } catch (error) {
+    if (error instanceof HTTPError && error.response.statusCode < 500) {
+      instagramConnectionState.on();
+      return;
+    }
```

Comment:
So if I use an incorrect Instagram secret in this call, the status code would be >= 500?

Reply:
Are we testing here the validity of our credentials or the status of the connection? I used the same logic in the LinkedIn credential to try to catch connection errors coming from the provider.

Reply:
In that case I couldn't find a way to use the secret outside the OAuth flow without getting an Unauthorized error.

Reply:
In case of at least one credential we were able to find a request that would produce different errors when the parameters we give it are bad and when we give it incorrect client_id. The former we singled out and ignored as expected error. The latter and all other errors we treated as connection failure. 

In other words, we need one of the following. 1) An endpoint which only requires a correct client_id and secret and fails if they are incorrect. 2) An endpoint where we can tell if it failed because of incorrect client_id or for some other reason.